src/lib/cfa/ntp_conf.rb

```diff
@@ -119,27 +111,27 @@ def initialize(augeas_tree)
       end
 
       def each
-        record_entries.each do |augeas_element|
-          yield Record.new_from_augeas(augeas_element)
+        record_entries.each do |record|
+          yield record
         end
```

Comment:
well, problem I have with this implementation is that it kills each ability to act like iterator ( I can give you some articles on this topic )

to demonstrate it try this example code and its output:

```
class C
  def initialize
    @arr = [1,2,3]
  end

  def each(&block)
    @arr.each(&block)
  end
end

o = C.new
o.each { |i| puts i.to_s }
puts o.each.count

class D
  def initialize
    @arr = [1,2,3]
  end

  def each(&block)
    @arr.each do |record|
      yield record
    end
  end
end

o = D.new
o.each { |i| puts i.to_s }
puts o.each.count
```